File: C#/TAP_Unidad03_Practica03/TAP_Unidad03_Practica03/Models/Calificaciones.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TAP_Unidad03_Practica03.Models
{
    class Calificaciones
    {
        int id = 0;
        Alumno alumno = null;
        float calificacion;
        Materia materia=null;

        public Calificaciones()
        {
            id = 0;
            alumno = new Alumno();
            calificacion = 0;
            materia = new Materia();
        }

        public int Id
        {
            get { return id; }
            set { id=value; }
        }

        public Alumno Alumno
        {
            get { return alumno; }
            set { alumno = value; }
        }

        public float Calificacion
        {
            get { return calificacion; }
            set { calificacion = value; }
        }

        public Materia Materia
        {
            get { return materia; }
            set { materia = value; }
        }

        public List<Calificaciones> listaMateria(List<Calificaciones> listaCalif,String nomMateria)
        {
            List<Calificaciones> lista = new List<Calificaciones>();
            foreach (Calificaciones indice in listaCalif)

            {
                if (indice.Materia.Nombre_materia == nomMateria)
                {
                    lista.Add(indice);
                }
            }

            return lista;
        }
    }
}

File: JAVA/TAP_Unidad03_Practica03/src/Models/Calificaciones.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;

import java.util.ArrayList;

/**
 *
 * @author Carlos J
 */
public class Calificaciones {
    
    private int id = 0;
    private Alumno alumno = null;
    private float calificacion;
    private Materia materia=null;

    public Calificaciones(){
        id = 0;
        alumno = new Alumno();
        calificacion = 0;
        materia = new Materia();
    }

    //Saca la lista de calificaciones por materia
    public ArrayList<Calificaciones> listaMateria(ArrayList<Calificaciones> listaCalif,String nomMateria){
        ArrayList<Calificaciones> lista = new ArrayList<>();
        
        for (Calificaciones indice:listaCalif){
            if (indice.getMateria().getNombre_materia().equalsIgnoreCase(nomMateria)){
                lista.add(indice);
            }
        }

        return lista;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Alumno getAlumno() {
        return alumno;
    }

    public void setAlumno(Alumno alumno) {
        this.alumno = alumno;
    }

    public float getCalificacion() {
        return calificacion;
    }

    public void setCalificacion(float calificacion) {
        this.calificacion = calificacion;
    }

    public Materia getMateria() {
        return materia;
    }

    public void setMateria(Materia materia) {
        this.materia = materia;
    }
}
